### PR TITLE
Session auth

### PR DESCRIPTION
**Pull request by allouis**
refs https://github.com/tryghost/ghost/issues/9865
requires https://github.com/TryGhost/Ghost/pull/9912
This updates the client to use the v2/admin api, and the session based authentication. It probably needs more tests, i haven't quite learnt how all the ember testing stuff works though. I've tested locally creating/editing posts, tags, settings and setting up a blog from scratch.

### CODE REVIEW

**Comment by allouis** [comment]
We think the tests are failing because of this change: https://github.com/TryGhost/Ghost-Admin/commit/4ba9c6e5824564689cb1812974a1da2fd35e57ae

File: tests/acceptance/authentication-test.js [hunk 1]
```diff
@@ -21,92 +19,6 @@ describe('Acceptance: Authentication', function () {
         destroyApp(application);
     });
 
-    describe('setup redirect', function () {
```
Comment (kevinansfield) [inline tests/acceptance/authentication-test.js]: I think this test is still required?

Comment (allouis) [reply tests/acceptance/authentication-test.js]: I think you're right :+1:

File: tests/acceptance/signin-test.js [hunk 2]
```diff
@@ -1,4 +1,3 @@
-import deparam from 'npm:deparam';
```
Comment (kevinansfield) [inline tests/acceptance/signin-test.js]: The two places this was used have been removed so we can fully remove this dependency.

**Comment by kevinansfield** [comment]
Code looks fine, there's not too much to test here because the API itself hasn't changed so the existing mirage setup is still fine.

Two other places I'm aware of that will need changes are the [labs](https://github.com/TryGhost/Ghost-Admin/blob/master/app/controllers/settings/labs.js#L131-L142) and [theme](https://github.com/TryGhost/Ghost-Admin/blob/master/app/controllers/settings/design.js#L173-L184) file download routines because they use query-param access tokens in an iframe to trigger browser downloads. I'm assuming we won't have to do anything special with them now because the request will have the cookie attached?

**Comment by allouis** [comment]
Sweet :relaxed: 

I just tested them and they still work :+1: So i'll remove the param from them!

**Comment by kevinansfield** [comment]
Awesome 😄 I'll do some manual testing later then 🙂 I'll hold off on merging until the session auth is merged in Ghost.

File: tests/unit/authenticators/cookie-test.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import CookieAuthenticator from 'ghost-admin/authenticators/cookie';
```
Comment (allouis) [inline tests/unit/authenticators/cookie-test.js]: Added authenticator tests here @kevinansfield  :relaxed:

File: tests/unit/authenticators/cookie-test.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import sinon from 'sinon';
+import {beforeEach, describe, it} from 'mocha';
+import {expect} from 'chai';
+
+describe.only('BaseAuthenticator', () => {
```
Comment (kevinansfield) [inline tests/unit/authenticators/cookie-test.js]: :trollface:

Comment (allouis) [reply tests/unit/authenticators/cookie-test.js]: LOL :sob:

**Pull request merged**